glom/core.py

```diff
@@ -1840,12 +1893,16 @@ def _glom(target, spec, scope):
     scope[Spec] = spec
     scope[UP] = parent
 
-    if isinstance(spec, TType):  # must go first, due to callability
-        return _t_eval(target, spec, scope)
-    elif callable(getattr(spec, 'glomit', None)):
-        return spec.glomit(target, scope)
-
-    return scope[MODE](target, spec, scope)
+    try:
+        if isinstance(spec, TType):  # must go first, due to callability
+            return _t_eval(target, spec, scope)
+        elif callable(getattr(spec, 'glomit', None)):
+            return spec.glomit(target, scope)
+
+        return scope[MODE](target, spec, scope)
+    except Exception:
+        scope[ROOT].setdefault(ERROR_SCOPE, scope)
```

Comment:
setdefault without assignment eh? I guess it's fewer characters than the alternative.